### PR TITLE
Minimum difference calculator improvements - part 1

### PR DESCRIPTION
**Pull request by razvanpapadopol**
Fixes part of https://github.com/Automattic/abacus/issues/756

* Fix the calculator to pass the correct min diff value so we no longer [divide it by 100](https://github.com/Automattic/abacus/pull/782#discussion_r984632842).
* Fixed precision issues when "applying min diff" for `Conversion` metric values (eg: 0.99pp from the calculator is applied as 0.989999pp).
* `Apply min diff` button is now disabled if min diff value is `0`.
* Prevent the case when no sample size is entered and the calculator displays `Infinity` in various places.
* Adds tests for the `MinDiffCalculator` component.

## How has this been tested?

- Automated tests that cover added/changed functionality
- Manual testing with production data (locally)
- Manual testing with mock data (locally or on Vercel)
- Additional manual testing instructions (please specify):
  - Test the precision issue and UX issues mentioned above.

### CODE REVIEW

**Review by jessie-ross — APPROVED** [review]
This is looking good, see comments below.

File: src/utils/math.ts [hunk 1]
```diff
@@ -88,3 +88,10 @@ export function samplesRequiredPerVariationForConversion(
   const zScore = 2.5
   return Math.ceil(Math.pow(zScore / delta, 2) * variance)
 }
+
+/**
+ * Round a number to at most nPlaces decimal places
+ */
+export function roundToNumPlaces(x: number, nPlaces: number): number {
```
[the remark this answers is not in the document]
Comment (jessie-ross) [reply src/utils/math.ts:95]: Actually maybe this is needed 🤔

[the remark this answers is not in the document]
Comment (razvanpapadopol) [reply src/utils/math.ts]: The lodash function seems like a good replacement and it tests well in all our precision corner cases. We'll be using it directly until we see the need for a specific abstraction.
